test-spinnaker-k8s.groovy

```diff
@@ -0,0 +1,48 @@
+node('spin-k8s-test') {
+    properties([parameters([
+            string(defaultValue: '72f988bf-86f1-41af-91ab-2d7cd011db47', description: 'Tenant id', name: 'tenant_id'),
+            string(defaultValue: 'AzDevOpsTestingSP', description: 'Credentials id', name: 'cred_id')
+            ])])
+
+    def venv_name= env.JOB_BASE_NAME + '-venv'
+    sh 'virtualenv ' + venv_name
+    def activate_venv = '. ' + env.WORKSPACE + '/' + venv_name + '/bin/activate;'
+
+    scenario_name = sh(returnStdout: true, script: 'echo "' + env.JOB_BASE_NAME + '$(uuidgen | sed \'s/-//g\')"').trim()
+
+    stage('Init') {
+        dir('devopsci') {
+            checkout scm
+        }
+        dir('spinnaker') {
+            git 'https://github.com/spinnaker/spinnaker.git'
+        }
+        dir('citestpackage') {
+            // Eventually this repo will be its own python package and we won't have to install it separately
+            git 'https://github.com/google/citest.git'
+            sh activate_venv + 'pip install -r requirements.txt'
+        }
+    }
+
+    stage('Deploy Quickstart Template') {
+        def script_path = 'devopsci/scripts/deploy-spinnaker-k8s.sh'
+        sh 'sudo chmod +x ' + script_path
+        withCredentials([usernamePassword(credentialsId: params.cred_id, passwordVariable: 'client_secret', usernameVariable: 'client_id')]) {
+            sh script_path + ' -sn ' + scenario_name + ' -ci ' + env.client_id + ' -cs ' + env.client_secret + ' -ti ' + params.tenant_id
+        }
+    }
+
+    stage('Run Test') {
+        sh 'ssh -F ' + scenario_name + '/ssh_config -f -N tunnel-start'
+        dir('spinnaker/testing/citest') {
+            sh activate_venv + 'pip install -r requirements.txt;PYTHONPATH=.:spinnaker python tests/kube_smoke_test.py --native_host=localhost'
+        }
+        sh 'ssh -O "exit" -F ' + scenario_name + '/ssh_config tunnel-stop'
+    }
+
+    stage('Clean Up') {
```

Comment:
The Clean Up stage will not run if any of the previous steps failed.
If this is your desired outcome (you need to look and see what went wrong) I recommend setting up a MS Teams notification.
You can find some sample code in src\com\microsoft\azure\devops\ci\utils.groovy : loadJoabProperties
We store the hook urls in the credentials store so you can store yours there too, or if you want to notify the DevOpsCI channel just use the credential id 'notification_hook_azure-vm-agents_jenkinci